### PR TITLE
[RNMobile] Blur removed rich text to prevent the crash on Android

### LINKED ISSUE #10491: IllegalViewOperationException: Unable to execute operation dispatchViewManagerCommand on view with tag: 159, since the view does...
Sentry Issue: [WORDPRESS-ANDROID-1TS](https://sentry.io/share/issue/132814a30e254edf9689f79eb92fb9c7/)

```
IllegalViewOperationException: Unable to execute operation dispatchViewManagerCommand on view with tag: 159, since the view does not exists
    at com.facebook.react.uimanager.UIImplementation.assertViewExists(UIImplementation.java:867)
    at com.facebook.react.uimanager.UIImplementation.dispatchViewManagerCommand(UIImplementation.java:747)
    at com.facebook.react.uimanager.UIManagerModule.dispatchCommand(UIManagerModule.java:660)
    at com.facebook.react.uimanager.UIManagerModule.dispatchViewManagerCommand(UIManagerModule.java:655)
    at java.lang.reflect.Method.invoke(Method.java)
...
(9 additional frame(s) were not displayed)

Unable to execute operation dispatchViewManagerCommand on view with tag: 159, since the view does not exists
```

### PR DESCRIPTION
**Pull request by lukewalczak**
## Description

Fixes: https://github.com/wordpress-mobile/WordPress-Android/issues/10491

gutenberg mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2918

Introduced `isNewRichTextAdded` flag to distinguish whether RichText was unmounted, because of adding new RichText via pressing `enter`. If `false` it indicates block is removed and RichText should be blurred.

Fixes also the keyboard which remained after removing RichText block.

## How has this been tested?

1. Open an app on Android
2. Create new draft
3. Add `Columns` block, choose `50/50`
4. Select first column
5. Add `Heading` block within column
6. Remove `Heading` block
7. Select `Columns` block
8. Open `Columns` settings
**Expect there is no crash**

before | after
--- | ---
crash_a | fix_crash_a

9. Add `Heading` block after `Columns`
10. Remove `Heading` block pressing `backspace` button on device keyboard
**Expect keyboard doesn't remain on screen**

before | after
--- | ---
key_a | fix_key_a

## Screenshots 

Attached above

## Types of changes

Bug fix
Crash fix

## Checklist:
- [X] My code is tested.
- [X] My code follows the WordPress code style. 
- [X] My code follows the accessibility standards. 
- [X] My code has proper inline documentation. 
- [X] I've included developer documentation if appropriate. 
- [X] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/rich-text/src/component/index.native.js
Comment (dratwas) [inline packages/rich-text/src/component/index.native.js:695]: Could you please add an info about the reason why we don't want to blur rich text when adding new one?

Comment (lukewalczak) [reply packages/rich-text/src/component/index.native.js]: Done, could you please test it?

**Review by Tug — APPROVED** [review]
LGTM 👍

File: packages/rich-text/src/component/index.native.js
Comment (dratwas) [reply packages/rich-text/src/component/index.native.js]: Thanks :)

File: packages/rich-text/src/component/index.native.js
Comment (dratwas) [inline packages/rich-text/src/component/index.native.js:114]: This is not a blocker but I think it should be `false` :)

**Review by dratwas — APPROVED** [review]
Left one minor comment. Everything works well, thanks @lukewalczak for this PR.

File: packages/rich-text/src/component/index.native.js
Comment (dratwas) [inline packages/rich-text/src/component/index.native.js:698]:
```suggestion
			return;
```

We don't need to return boolean in `componentWillUnmount`

**Review by dratwas — CHANGES_REQUESTED** [review]
Hmmm, after some time I think that maybe we should not merge this yet? I'm not sure if it is correct because we assume that we always add a new block `onEnter` which is maybe true at this moment but I believe it could be different in the future since we call `onEnter` from props. So from the rich-text component perspective, we do not know what happens `onEnter`. Additional thing is that this PR adds some new logic to the rich-text which is already a bit too complex.

Also because of this change e2e tests are failing (image)

maybe we could always call the `editor.blur` in `componentWillUnmount` and fix the issue related to adding a new rich-text by using `return/enter` key (It creates new blocks and replaces the old one. The old one is unmounted the editor.blur is called and keyboard hides). 
Instead, we could use the same `clientId` for the new block, and thanks to that the `componentWillUnmount` will not be called when adding the new block with `return/enter` key, and the issue with the keyboard will be also solved :)

**Review by Tug — APPROVED** [review]
Fix is much simpler now even if it doesn't address the keyboard "jumpiness" issue. This worked great in my testing, I could not reproduce the crash 🎉 
Thanks for the fix @lukewalczak let's merge this and open an issue for the keyboard problem

**Review by dratwas — APPROVED** [review]

**Pull request merged**
